app/services/permissions/order.rb

```diff
@@ -2,24 +2,26 @@
 
 module Permissions
   class Order
-    def initialize(user)
+    def initialize(user, params = nil)
```

Comment:
we should only send params down to service classes if really needed. This is not the case. We dont need the full params object here, only params[:q]. I'd say you need to pass params[:q] and give it a good name at this service class level.
This params[:q] is specific to ransack, right? Maybe order_search_params or ransack_order_search_params?
This will make
def search_orders?
  order_search_params.present?
end

Reply:
Yes, I need to tidy up a bit.